02-r-programming.Rmd

```diff
@@ -240,88 +265,107 @@ model$coefficients
 Here's a function that returns a list.
 
 ```{r}
-sum_multiply <- function(x,y){
+sum_multiply <- function(x,y) {
   sum <- x + y
-  product <- x*y
+  product <- x * y
   list("Sum" = sum, "Product" = product)
 }
 
-result <- sum_multiply(2,3)
+result <- sum_multiply(2, 3)
 result$Sum
 result$Product
 ```
 
 ## Data frames
 
-R is an old programming language.  The original `data.frame` object has been updated with the newer and better `tibble` (like the word "table").  **Tibbles are really lists of vectors, where each column is a vector**.  
+You can think of a data frame as a table that is implemented as a list of vectors.
 
 ```{r}
-#the tibble library has functions for making tibbles
+df <- data.frame(
+  age = c(25, 35),
+  has_fsa = c(FALSE, TRUE)
+)
+df
+```
+
+You can also work with tibbles, which are data frames but have nicer printing:
+
+```{r}
+# The tibble library has functions for making tibbles
 library(tibble) 
-data <- tibble(age = c(25, 35), has_fsa = c(F, T))
-data
+df2 <- tibble(
+  age = c(25, 35), has_fsa = c(FALSE, TRUE)
+)
+df2
 ```
 
 To index columns in a tibble, the same "$" is used as indexing a list.
 
 ```{r}
-data$age
+df2$age
 ```
 
 To find the number of rows and columns, use `dim`.
 
 ```{r}
-dim(data)
+dim(df2)
 ```
 
-To fine a summary, use `summary`
+To find a summary, use `summary`
 
 ```{r}
-summary(data)
+summary(df2)
 ```
 
 ## Pipes
 
-The pipe operator `%>%` is a way of making code *modular*, meaning that it can be written and executed in incremental steps.  Those familiar with Python's Pandas will be see that `%>%` is quite similar to ".".  This also makes code easier to read.
+The pipe operator `%>%` is a way of making code *modular*, meaning that it can 
+be written and executed in incremental steps.  Those familiar with Python's Pandas 
+will be see that `%>%` is quite similar to ".".  This also makes code easier to 
+read.
 
 In five seconds, tell me what the below code is doing.
 
 ```{r}
 log(sqrt(exp(log2(sqrt((max(c(3, 4, 16))))))))
 ```
 
-Getting to the answer of 1 requires starting from the inner-most nested brackets and moving outwards from right to left.
+Getting to the answer of 1 requires starting from the inner-most nested brackets
+and moving outwards from right to left.
 
 The math notation would be slightly easier to read, but still painful.
 
 $$log(\sqrt{e^{log_2(\sqrt{max(3,4,16)})}})$$
 
-Here is the same algebra using the pipe.  To read this, replace the `%>%` with the word `THEN`.
+Here is the same algebra using the pipe.  To read this, replace the `%>%` with 
+the word `THEN`.
 
 ```{r message = F, warning = F}
-#the pipe is from the dplyr library
-library(dplyr) 
+# The pipe is from the magrittr library
```

Comment:
This isn't that important, as I'm guessing tidyverse is one of the SOA included packages and users can just `library(tidyverse)`?